### PR TITLE
Add the ability to specify environment variables via flags

### LINKED ISSUE #107: Suggested option: --env-file
When I create systemd unit files, I sometimes use the `EnvironmentFile` directive so that environment config variables can be stored separately from the unit file.  Similarly in sys-v init scripts one can "dot source" a file to include its environment.   How about a `--env-file` option for pleaserun that adds the relevant line to the init script/config?

### PR DESCRIPTION
**Pull request by josegonzalez**
- You can specify the --environment-variables flag 1 or more times to set multiple environment variables
- You can import environment variables from a file via the --environment-file flag
- Environment variables are written to an /etc/default/NAME file in the format needed for a given init program, with a few exceptions
  - runit uses an embedded env call
  - launchd has it written in the plist file
  - systemd-user does not implement env var support
  - upstart 0.6.5 does not implement env var support, though the file is created

The sysconfig file is left untouched.

Refs #9
Closes #94
Closes #107

### CODE REVIEW

**Comment by josegonzalez** [comment]
@jordansissel friendly *bump*

**Comment by jordansissel** [comment]
Reviewing now

File: lib/pleaserun/cli.rb [hunk 1]
```diff
@@ -44,9 +44,10 @@ class PleaseRun::CLI < Clamp::Command # rubocop:disable ClassLength
 
     # Turn the attribute name into a flag.
     option "--#{facet.name.to_s.gsub("_", "-")}", facet.name.to_s.upcase, facet.description,
-           :attribute_name => facet.name
+           :attribute_name => facet.name,
+           :multivalued => facet.options.fetch(:multivalued, false)
```
Comment (jordansissel) [inline lib/pleaserun/cli.rb]: :+1:

File: lib/pleaserun/platform/base.rb [hunk 2]
```diff
@@ -210,4 +232,18 @@ def log_path_stdout
...
   end
+
+  def parsed_environment_variables
+    return {} if environment_file.nil?
+    return {} unless File.exist?(environment_file)
+    Dotenv::Parser.call(File.open(environment_file, "rb:bom|utf-8", &:read))
+  end
+
+  def all_environment_variables
+    variables = parsed_environment_variables.merge(environment_variables)
```
Comment (jordansissel) [inline lib/pleaserun/platform/base.rb]: `rspec` fails here:

```
  An error occurred in an after hook
    TypeError: no implicit conversion of nil into Hash
    occurred at /home/jls/projects/pleaserun/lib/pleaserun/platform/base.rb:243:in `merge'
```

Comment (jordansissel) [inline lib/pleaserun/platform/base.rb]: I had to refresh myself with the whole facet things. Some tests have `environment_variables` defaulted to {} some have it `nil` ... super weird. I'll look into it.

**Comment by jordansissel** [comment]
Big +1 to the idea. I'd like the specs broken by this PR to be passing (master has some failing specs I need to fix).

I'm open to merging this with some successful hand-testing and I can attend to the specs later. Will test soon, but welcome any other feedback. <3

File: lib/pleaserun/platform/base.rb (hunk 2 above)
Comment (josegonzalez) [inline lib/pleaserun/platform/base.rb]: Let me just do gentle checking to ensure the code doesn't do something silly here.

**Comment by josegonzalez** [comment]
Hopefully fixed the spec issues!

**Comment by jordansissel** [comment]
Seems OK to me :)

**Comment by josegonzalez** [comment]
Awh yeah can I get a release? :D

**Pull request merged**